lib/ansible/module_utils/common/file.py

```diff
@@ -158,7 +158,7 @@ def set_lock(self, path, tmpdir, lock_timeout=None):
 
         self.lockfd = open(lock_path, 'w')
 
-        if lock_timeout <= 0:
+        if lock_timeout <= 0 and lock_timeout is not None:
```

Comment:
This needs to be reversed.

```suggestion
        if lock_timeout is not None and lock_timeout <= 0:
```

```pycon
>>> lock_timeout = None
>>> lock_timeout <= 0 and lock_timeout is not None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<=' not supported between instances of 'NoneType' and 'int'
>>> lock_timeout is not None and lock_timeout <= 0
False
```

Reply:
Ah, yes, was testing just with Python 2 for some reason.